vignettes/pull_request_comments.Rmd

```diff
@@ -0,0 +1,276 @@
+The process should not take more than 2 minutes.
+
+The token should then be copied and pasted into `../.ssh/github_token`.
+Both the `.ssh` folder and `github_token` file may need to be created
+manually. Check for hidden folders before creating the `.ssh` folder.
+The `github_token` file should not have any file extensions.
+
+```{r Scan GitHub Token}
+# your file github_token (a text file) contains the GitHub token API
+token <- scan("~/.ssh/github_token",what="character",quiet=TRUE)
+```
+
+## GitHub Pull Request Comments
+
+In GitHub, a Pull Request conversation starts off with the Pull Request
+description. Pull Request comments are made by users. A reviewer can
+start a review where they can add a single comment, request changes, or
+approve for merging. This starts the review comment. From there, in-line
+code comments can be made from the review.
+
+# Download GitHub Pull Request Comments
+
+## Conversation Comments
+
+### Pull Request Comments
+
+Pull Request Comments not connected to a review, appear in the GitHub
+Pull Request Conversations tab but is retrieved by the issue comments
+endpoint. Using this endpoint, both issues and pull request comments are
+retrieved.
+
+```{r Collect comments by pull request number}
+#gh call but with date
+# get the data
+gh_response_issue_or_pr_comment <- github_api_project_issue_or_pr_comment_refresh(owner,repo,token,save_path_issue_or_pr_comments,verbose=TRUE)
+
+# create directory and iterate over data
+#dir.create(save_path_issue_or_pr_comments)
+github_api_iterate_pages(token,gh_response_issue_or_pr_comment,
+                         save_path_issue_or_pr_comments,
+                         prefix="issue_or_pr_comment",
+                         verbose=TRUE)
+```
+
+It is important to filter out the issue comments in this notebook. The
+key to this is in the html_url of each comment, differentiated by
+/issues or /pull
+
+```{r}
+pr_comments <- lapply(list.files(save_path_issue_or_pr_comments, full.names = TRUE),read_json)
+pr_comments <- lapply(pr_comments, github_parse_project_issue_or_pr_comments)
+pr_comments <- rbindlist(pr_comments,fill=TRUE)
+
+#filter only pull requests using grepl
+pr_comments <- pr_comments[grepl("/pull/", html_url)]
+
+head(pr_comments,10)  %>%
+  gt(auto_align = FALSE) 
+```
+
+### First Pull Request Comment
+
+We may notice the first comment of the pull request is missing. The pull
+request description is treated as a pull request comment in the
+conversations tab by GitHub. However, it is not retrievable by the other
+comments endpoints. To get this, we must utilize the pull request
+endpoint to gather information on the pull request, and retrieve the
+body paragraph for that pull request.
+
+```{r Collect information on pull request}
+
+gh_response <- github_api_project_pull_request_refresh(owner, repo, token, save_path_pull_request)
```

Comment:
This will not work due to the inner function reliance on a timestamp being passed to a endpoint that does not have a timestamp as parameter. This function needs to be replaced by another to download PRs.